2-projects/1-todo-list/script.js

```diff
@@ -12,13 +42,34 @@ let todos = [
 
 populateTodoList(todos);
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+
+ 
+  const inputField = document.getElementById("todoInput");
+  const newTodoText = inputField.value;
+
+  
+  const newTodo = {
+    task: newTodoText,
+    completed: false
+  };
+
+  
+  todos.push(newTodo);
+
+ 
+  populateTodoList(todos);
+
+
+  inputField.value = "";
```

Comment:
This is really good work - I like the fact that you have added the new todo to the list of todos and then run the `populateTodoList()` function again! Note how it means that all your todos are rendered in exactly the same way, with the same event handlers set up on them. React actually works in a similar way to this, but manages to get round some of the problems with it (such as changes to the DOM in the browser being relatively slow!). Encouraging stuff :)